shared/actions/tracker.js

```diff
@@ -296,10 +316,34 @@ export function onFollow (username: string, localIgnore: bool): (dispatch: Dispa
   }
 }
 
-export function onClose (username: string): Action {
-  return {
-    type: Constants.onClose,
-    payload: {username}
+function _dismissWithToken (trackToken) {
+  const params : track_dismissWithToken_rpc = {
+    method: 'track.dismissWithToken',
+    param: {trackToken},
+    incomingCallMap: {},
+    callback: err => {
+      if (err) {
+        console.log('err dismissWithToken', err)
+      }
+    }
+  }
+  engine.rpc(params)
+}
+
+export function onClose (username: string): TrackerActionCreator {
+  return (dispatch, getState) => {
+    const trackToken = _getTrackToken(getState, username)
+
+    if (trackToken) {
```

Comment:
if we can dismiss now, let's do so. otherwise onClose will bookkeep for us that we need to later. this is **screaming** for a saga